server/api/services/project.ts

```diff
@@ -73,3 +82,41 @@ export const updateProjectById = async (
 
 	return result;
 };
+
+export const validUniqueServerAppName = async (appName: string) => {
```

Comment:
@hehehai since you just validate and throw an error, what do you think of this proposal?

```ts
await validUniqueServerAppNameOrThrow(input.appName);

export const validUniqueServerAppNameOrThrow = async (appName?: string) => {
  if (!appName) return;

  const query = await db.query.projects.findMany({
    with: {
      applications: {
        where: eq(applications.appName, appName),
      },
      mariadb: {
        where: eq(mariadb.appName, appName),
      },
      mongo: {
        where: eq(mongo.appName, appName),
      },
      mysql: {
        where: eq(mysql.appName, appName),
      },
      postgres: {
        where: eq(postgres.appName, appName),
      },
      redis: {
        where: eq(redis.appName, appName),
      },
    },
  });

  // Filter out items with non-empty fields
  const nonEmptyProjects = query.filter(
    (project) =>
      project.applications.length > 0 ||
      project.mariadb.length > 0 ||
      project.mongo.length > 0 ||
      project.mysql.length > 0 ||
      project.postgres.length > 0 ||
      project.redis.length > 0
  );

  if (nonEmptyProjects.length > 0) {
    throw new TRPCError({
      code: "CONFLICT",
      message: "Service with this 'AppName' already exists",
    });
  }
};
```